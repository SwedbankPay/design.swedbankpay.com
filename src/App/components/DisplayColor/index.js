import React from "react";
import PropTypes from "prop-types";

const ColorDisplay = ({ hex, rgb, fontColor, border }) => (
    <div className={`${border ? border : ""}`}>
        <svg>
            <rect fill={hex} />
        </svg>
        <div className={`pl-3 ${fontColor ? fontColor : ""}`}>
            <p className="mb-0">{hex}</p>
            <p>{rgb}</p>
        </div>
    </div>
);

ColorDisplay.propTypes = {
    hex: PropTypes.string.isRequired,
    rgb: PropTypes.string.isRequired,
    fontColor: PropTypes.string,
    border: PropTypes.string
};

export default ColorDisplay;
