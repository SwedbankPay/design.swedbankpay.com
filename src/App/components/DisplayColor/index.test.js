import React from "react";
import { shallow } from "enzyme";

import DisplayColor from "./index";

describe("DisplayColor", () => {
    it("is defined", () => {
        expect(DisplayColor).toBeDefined();
    });

    it("renders with required props", () => {
        const colors =
            {
                hex: "#FDC129",
                rgb: "253, 193, 41"
            };

        const wrapper = shallow(<DisplayColor {...colors}/>);

        expect(wrapper).toMatchSnapshot();
    });

    it("alert if required properties is not provided", () => {
        const colors =
            {
                fontColor: "512B2B",
                border: "has-border"
            };

        console.error = jest.fn();

        shallow(<DisplayColor {...colors}/>);

        expect(console.error).toHaveBeenCalled();
    });

    it("renders with optional props", () => {
        const colors =
            {
                hex: "#FDC129",
                rgb: "253, 193, 41",
                fontColor: "512B2B",
                border: "has-border"
            };

        const wrapper = shallow(<DisplayColor {...colors}/>);

        expect(wrapper).toMatchSnapshot();
    });
});
