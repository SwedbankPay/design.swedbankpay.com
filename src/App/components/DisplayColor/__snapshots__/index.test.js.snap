// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayColor renders with optional props 1`] = `
<div
  className="has-border"
>
  <svg>
    <rect
      fill="#FDC129"
    />
  </svg>
  <div
    className="pl-3 512B2B"
  >
    <p
      className="mb-0"
    >
      #FDC129
    </p>
    <p>
      253, 193, 41
    </p>
  </div>
</div>
`;

exports[`DisplayColor renders with required props 1`] = `
<div
  className=""
>
  <svg>
    <rect
      fill="#FDC129"
    />
  </svg>
  <div
    className="pl-3 "
  >
    <p
      className="mb-0"
    >
      #FDC129
    </p>
    <p>
      253, 193, 41
    </p>
  </div>
</div>
`;
